<!-- -->
<launch>

  <!-- carla parameter -->
  <arg name='host' default='localhost'/> <!-- host the carla server is running on -->
  <arg name='port' default='2000'/> <!-- port of the carla server -->
  <arg name="town" default="Town01"/> 
  <arg name='timeout' default='10'/> <!-- the carla timeout for server communication -->
  <arg name='passive' default='False'/> <!-- enable/disable passive mode -->
  <arg name='synchronous_mode_wait_for_vehicle_control_command' default='False'/><!-- should the ros bridge wait for a vehicle control command before proceeding with the next tick -->
  <arg name='synchronous_mode' default='True'/><!-- enable/disable synchronous mode -->
  <arg name='fixed_delta_seconds' default='0.05'/><!-- frequency of the carla ticks -->
  <arg name='register_all_sensors' default='False'/>
  
   
  <arg name="role_name" default="ego_vehicle"/> <!-- role name of the ego vehicle (used for identification in many nodes) -->
  <arg name='spawn_point' default='127.4,-195.4,2,0,0,180'/> 
  <!-- <arg name='spawn_point' default=""/>  -->
  <!-- <arg name='spawn_point' default='71.1,-45.7,35.0,0.0,0.0,180.0'/> -->
  <!-- <arg name="goal_point" default="-64.6,24.4,0.59,0.0,0.0,180"/> goal point for the ego vehicle -->

  <arg name="target_speed" default="8.33"/> <!-- target speed in m/s that the agent tries to follow -->
  <arg name="avoid_risk" default="True"/> <!-- should the agent avoid crashs and respect red traffic lights? -->

  <!-- ===================================================== -->

  <!-- Publish the target speed -->
  <node pkg="rostopic" type="rostopic" name="rostopic" args="pub /carla/$(arg role_name)/target_speed std_msgs/Float64 $(arg target_speed)"/>
  

  <arg name="control_time_step" default="0.1"/>

  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find control)/config/carla_viz.rviz" output="screen"/>

  <node pkg="control" type="global_planner.py" name="global_path_publisher" output="screen">
    <param name="role_name" value="$(arg role_name)" />
    <param name="host" value="$(arg host)" />
    <param name="port" value="$(arg port)" />
    <param name="timeout" value="$(arg timeout)" />
  </node>

  <node pkg="control" type="carla_bridge.py" name="controller" output="screen">
  </node>

</launch>
